rpcs3/Emu/RSX/GL/GLTexture.cpp

```diff
@@ -319,6 +319,14 @@ namespace rsx
 				std::vector<gsl::byte> staging_buffer)
 			{
 				int mip_level = 0;
+				if (is_compressed_format(format))
+				{
+					//Compressed formats have a 4-byte alignment
+					//TODO: Verify that samplers are not affected by the padding
+					width = align(width, 4);
+					height = align(height, 4);
```

Comment:
The endian issue is fixed now.

About the Vulkan problem I mentioned. There is that second patch I did which makes fragment_texture::get_exact_mipmap_count() return values >= 1. This is still needed with the new alignment code, otherwise the function returns 0 for the 512x2 texture and crashes (Vulkan) / misbehaves (OpenGL).